src/main/shared/bullet/bullet-store.ts

```diff
@@ -3,7 +3,7 @@ import { Bullet } from './bullet-model';
 import { Store } from '../store/store';
 
 @Singleton
-export class BulletStore extends Store<Bullet> {
+export class BulletStore<T extends Bullet = Bullet> extends Store<T> {
```

Comment:
not sure if this is needed but we will see

Reply:
Not sure about the best one